autoprotocol_utilities/container_helpers.py

```diff
@@ -1025,16 +1025,36 @@ def next_wells(plate, num=1, columnwise=False):
     Returns
     -------
 
-    genarator:
+    generator:
         This function will iteratively generate the next set of
         wells from the plate. Get the next plates using
         `next(generator)`. Wells will be returned as a
         WellGroup.
 
+    Raises
+    ------
+    ValueError
+        If plate is not a Container or a list of Containers
+    RumtimeError
+        If all wells have been iterated through
     '''
-
     next_index = 0
-    well_list = plate.all_wells(columnwise=columnwise)
-    while next_index <= plate.container_type.well_count - num:
+    assert isinstance(plate, (list, Container)), ("plate must be a Container "
+                                                  "or a list of Containers")
+    if isinstance(plate, list):
+        for p in plate:
+            assert isinstance(p, Container), ("all elements of plate "
+                                              "must be Containers")
+    if isinstance(plate, Container):
```

Comment:
Instead of having these two `isinstance` checks, why not move this one up and then remove the other check. Essentially always treat is as an array of plates?